tough/src/editor/mod.rs

```diff
@@ -225,30 +301,56 @@ impl RepositoryEditor {
             .context(error::PathUtf8 { path: target_path })?
             .to_owned();
 
-        self.add_target(target_name, target);
-        Ok(self)
+        Ok((target_name, target))
     }
 
-    /// Add a list of target paths to the repository
-    ///
-    /// See the note on `add_target_path()` regarding performance.
-    pub fn add_target_paths<P>(&mut self, targets: Vec<P>) -> Result<&mut Self>
-    where
-        P: AsRef<Path>,
-    {
-        for target in targets {
-            self.add_target_path(target)?;
-        }
+    /// Remove all targets from this repo
+    pub fn clear_targets(&mut self) -> Result<&mut Self> {
+        self.targets_editor_mut()?.clear_targets();
         Ok(self)
     }
 
-    /// Remove all targets from this repo
-    pub fn clear_targets(&mut self) -> &mut Self {
-        self.existing_targets
-            .get_or_insert_with(HashMap::new)
-            .clear();
-        self.new_targets.get_or_insert_with(HashMap::new).clear();
-        self
+    #[allow(clippy::too_many_arguments)]
+    /// Delegate target with name.
```

Comment:
Let's expand this docstring. In what delegated targets situation should I use this method?